### PR TITLE
Simplify chemml so that 3D coords are not used

### PR DESCRIPTION
**Pull request by psavery**
Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

### CODE REVIEW

**Comment by psavery** [comment]
This PR depends on [this PR](https://github.com/OpenChemistry/openchemistrypy/pull/71) (and its dependent PR) to work correctly. Some logic still needs to be improved on the mongochemserver side for calculation outputs that do not have 3D coordinates.

@cryos, is it acceptable to call the output from chemml here `cjson` if it doesn't have any keys other than the `properties` (the results of the calculation)? I could potentially also make it into just a `json` output or something else.

**Comment by cryos** [comment]
I think if the output is structured such that it has the chemicalJson key, and smiles or inchi key then properties object it is a CJSON. It is also OK to just output JSON as CJSON may expect 3D structure.

File: docker/chemml/src/run.py [hunk 1]
```diff
@@ -44,7 +16,10 @@ def run_calculation(geometry_file, output_file):
         "density": pred[2]
     }
 
-    cjson.setdefault('properties', {}).update(properties)
+    # We will only output the properties
+    cjson = {
```
Comment (cryos) [inline docker/chemml/src/run.py]: I would repeat the smiles in the output. Once we support lists of smiles I think something of the form
`
[ { 'smiles': smiles, 'properties': properties } ]
`
Would work and the output ingest can then unroll the outputs.

**Comment by psavery** [comment]
With [this PR](https://github.com/OpenChemistry/mongochemserver/pull/121) along the other dependent PR's mentioned above, I can run chemml without ever using 3D coordinates :-). The calculation runs successfully and prints out the properties in the notebook, with no errors.

3D coordinates, however, still get generated automatically when the mol gets created with `find_structure()` or `import_structure()`. If we make the 3D coordinates optional there, we can skip the 3D generation entirely for some calculations such as chemml.

The cjson that is stored in the calculation object as a result of all of this does not have 3D coordinates. Some parts of our codes (not called as a part of this chemml taskflow) may expect there to be 3D coordinates, so that may require a little refactoring.

Also, we will need to consider what to do about the rendering of structures with no 3D. Currently, it just doesn't render anything if you look at the calculation result on the client.
Screenshot from 2019-04-01 14-05-33

**Comment by psavery** [comment]
If you use [this PR](https://github.com/OpenChemistry/openchemistrypy/pull/79) along with all of the other ones (about 6 total I think), you can do `import_structure` without generating 3D coordinates, and run it through chemml and get the output, without ever generating 3D coordinates in any of the steps.

**Comment by psavery** [comment]
So, you can do things like this currently:
```
structure_smiles = ['CN1CNCN(C1)c1ccc(c2c1cccc2)C1NCNCN1', 'SC1CSCC(S1)(S)c1cccnc1', 'c1ccc(nc1)c1scc(c1)c1ccnc(c1)c1ccc2c(c1)cccc2']
mols = []
results = []
for smiles in structure_smiles:
    mols.append(oc.import_structure(smiles, gen3d=False))

for mol in mols:
    results.append(mol.calculate('psavery/chemml:latest', {}))

x = []
y = []
for result in results:
    x.append(result.data()['properties']['polarizability'])
    y.append(result.data()['properties']['refractiveIndex'])
```

And 3D coordinates are never generated. But I need to make it so that chemml can accept multiple molecules so it can run them all in one go, rather than having to spin it up for each molecule.

A question: is it acceptable to import all of the SMILES into the database before running chemml on all of them? When we import them into the database, they each do extra things such as generating an InChI, and an InChI key, and getting some properties from Avogadro, but they do not generate 3D coordinates. Is this acceptable, or do we want to pass SMILES directly into chemml without ever putting the structures in the database?

**Pull request merged**